File: src/260.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

class Solution {
   public:
    vector<int> singleNumber(vector<int>& nums) {
        vector<int> retVal(2, 0);

        unsigned int xorSum = 0;
        for (int num : nums) {
            xorSum ^= num;
        }
        unsigned int lsb = xorSum & (-xorSum);

        for (int num : nums) {
            retVal[((lsb & num) ? 1 : 0)] ^= num;
        }

        return retVal;
    }
};

int main(int argc, char** argv) {
    struct subject {
        vector<int> nums;
        int target;
    };
    vector<subject> testData{{{1, 2, 1, 3, 2, 5}}, {{-1, 0}}, {{0, 1}}};
    int numberOfTestCase = testData.size();
    /* Example
     *  Input: nums = [1,2,1,3,2,5]
     *  Output: [3,5]
     *
     *  Input: nums = [-1,0]
     *  Output: [-1,0]
     *
     *  Input: nums = [0,1]
     *  Output: [1,0]
     */

    Solution cSolution;
    vector<int> answer;
    for (int i = 0; i < numberOfTestCase; ++i) {
        cout << "Input: nums = [";
        for (long unsigned int j = 0; j < testData[i].nums.size(); ++j) {
            cout << ((j == 0) ? "" : ",") << testData[i].nums[j];
        }
        cout << "]\n";

        answer = cSolution.singleNumber(testData[i].nums);
        cout << "Output: [";
        for (long unsigned int j = 0; j < answer.size(); ++j) {
            cout << ((j == 0) ? "" : ",") << answer[j];
        }
        cout << "]\n";

        cout << "\n";
    }

    return EXIT_SUCCESS;
}

File: src/260.py
import sys
import logging
from typing import List, Optional
from collections import defaultdict, deque, Counter
from functools import cache, lru_cache
from bisect import bisect_left, bisect_right
from itertools import accumulate
from heapq import heapify, heappush, heappop


def logging_setting():
    LOG_LEVEL = logging.INFO    # DEBUG, INFO, WARNING, ERROR, CRITICAL
    if __debug__:
        LOG_LEVEL = logging.DEBUG

    logging.basicConfig(
        stream=sys.stderr,
        level=LOG_LEVEL,
        format="%(levelname)-6s %(asctime)s %(module)s:%(funcName)s:%(lineno)-4s %(message)s",
        datefmt='%Y/%m/%d %H:%M:%S')


class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        retVal = [0, 0]

        xorSum = 0
        for num in nums:
            xorSum ^= num
        lsb = xorSum & (-xorSum)

        for num in nums:
            idx = 0 if lsb & num == 0 else 1
            retVal[idx] ^= num

        return retVal


if __name__ == "__main__":
    logging_setting()

    try:
        logging.info("sys.version: %s", sys.version)
        print()

        pSolution = Solution()
        for nums in [[1, 2, 1, 3, 2, 5], [-1, 0], [0, 1]]:
            # /* Example
            #  *  Input: nums = [1,2,1,3,2,5]
            #  *  Output: [3,5]
            #  *
            #  *  Input: nums = [-1,0]
            #  *  Output: [-1,0]
            #  *
            #  *  Input: nums = [0,1]
            #  *  Output: [1,0]
            #  */
            logging.info("Input: nums = %s", nums)

            retVal = pSolution.singleNumber(nums)
            logging.info("Output: %s", retVal)

            print()
    except KeyboardInterrupt as exception:
        logging.error("%s: %s", exception.__class__.__name__, exception, exc_info=True)
        pass
    except Exception as exception:
        logging.error("%s: %s", exception.__class__.__name__, exception, exc_info=True)
        pass

    sys.exit(0)
